packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -47,8 +47,215 @@ def _create_file(
+    configuration_metadata = get_configuration_metadata_by_index(metadata, service_class)
+    output_metadata = get_output_metadata_by_index(metadata)
+
+    configuration_parameters_with_type_and_default_values, measure_api_parameters = (
+        get_configuration_parameters_with_type_and_default_values(
+            configuration_metadata, built_in_import_modules
+        )
+    )
+    output_parameters_with_type = get_output_parameters_with_type(
+        output_metadata, built_in_import_modules, custom_import_modules
+    )
+
+    _create_file(
+        template_name="measurement_plugin_client.py.mako",
+        file_name=f"{module_name}.py",
+        directory_out=directory_out_path,
+        class_name=class_name,
+        display_name=metadata.measurement_details.display_name,
+        configuration_metadata=configuration_metadata,
+        output_metadata=output_metadata,
+        service_class=service_class,
+        configuration_parameters_with_type_and_default_values=configuration_parameters_with_type_and_default_values,
+        measure_api_parameters=measure_api_parameters,
+        output_parameters_with_type=output_parameters_with_type,
+        built_in_import_modules=to_ordered_set(built_in_import_modules),
+        custom_import_modules=to_ordered_set(custom_import_modules),
+    )
+
+    print(
+        f"The measurement plug-in client for the service class '{service_class}' has been created successfully."
+    )
+
+
+def _create_client(
+    channel_pool: GrpcChannelPool,
+    discovery_client: DiscoveryClient,
+    built_in_import_modules: List[str],
+    custom_import_modules: List[str],
+    measurement_service_class: List[str] = [],
+    all: bool = False,
+    module_name: Optional[str] = "",
+    class_name: Optional[str] = "",
+    directory_out: Optional[str] = "",
+    interactive_mode: bool = False,
+) -> None:
+
+    measurement_service_class = _get_measurement_service_class(
+        all, interactive_mode, measurement_service_class, discovery_client
+    )
+
+    directory_out_path = _resolve_output_directory(directory_out)
+
+    is_multiple_client_generation = len(measurement_service_class) > 1
+    for service_class in measurement_service_class:
+        module_name, class_name = _get_class_and_module_names(
+            service_class, is_multiple_client_generation, module_name, class_name, interactive_mode
+        )
+        _generate_measurement_client(
+            discovery_client,
+            channel_pool,
+            service_class,
+            built_in_import_modules,
+            custom_import_modules,
+            module_name,
+            class_name,
+            directory_out_path,
+        )
+
+
 @click.command()
 @click.argument("measurement_service_class", nargs=-1)
+@click.option(
+    "-a",
+    "--all",
+    is_flag=True,
+    help="Creates Python Measurement Plug-In Client for all the registered measurement services.",
+)
```

Comment:
Does this need to be moved?

Reply:
I rearranged such that it matches the method signature.
```
def create_client(
    measurement_service_class: List[str],
    all: bool,
    module_name: Optional[str],
    class_name: Optional[str],
    directory_out: Optional[str],
    interactive: bool,
) -> None:
```

Reply:
Since this issue is minor and would only result in a slight increase in time to address, I’ll let it slide this time and avoid raising a separate PR for it. Moving forward, I’ll ensure that similar changes are handled more promptly.